File: app/Models/Issue.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Issue extends Model
{
    protected $fillable = [
        'integration_id',
        'platform',
        'platform_issue_id',
        'repository',
        'title',
        'description',
        'type',
        'state',
        'priority',
        'author_username',
        'author_avatar',
        'assignees',
        'labels',
        'comments_count',
        'due_date',
        'created_at_platform',
        'updated_at_platform',
        'closed_at',
    ];

    protected function casts(): array
    {
        return [
            'assignees' => 'array',
            'labels' => 'array',
            'due_date' => 'datetime',
            'created_at_platform' => 'datetime',
            'updated_at_platform' => 'datetime',
            'closed_at' => 'datetime',
        ];
    }

    /**
     * Get the integration that owns the issue.
     */
    public function integration()
    {
        return $this->belongsTo(Integration::class);
    }
}

File: bugradar_mobile/lib/models/issue.dart
class Issue {
  final int id;
  final int integrationId;
  final String platformIssueId;
  final String title;
  final String description;
  final String status;
  final String type;
  final String priority;
  final String url;
  final String? repository;
  final String? author;
  final List<String> labels;
  final DateTime createdAt;
  final DateTime updatedAt;
  final DateTime? closedAt;
  final Map<String, dynamic>? metadata;

  Issue({
    required this.id,
    required this.integrationId,
    required this.platformIssueId,
    required this.title,
    required this.description,
    required this.status,
    required this.type,
    required this.priority,
    required this.url,
    this.repository,
    this.author,
    required this.labels,
    required this.createdAt,
    required this.updatedAt,
    this.closedAt,
    this.metadata,
  });

  factory Issue.fromJson(Map<String, dynamic> json) {
    return Issue(
      id: json['id'] as int,
      integrationId: json['integration_id'] as int,
      platformIssueId: json['platform_issue_id'] as String,
      title: json['title'] as String,
      description: json['description'] as String? ?? '',
      status: json['status'] as String,
      type: json['type'] as String,
      priority: json['priority'] as String,
      url: json['url'] as String,
      repository: json['repository'] as String?,
      author: json['author'] as String?,
      labels: (json['labels'] as List<dynamic>?)?.map((e) => e.toString()).toList() ?? [],
      createdAt: DateTime.parse(json['created_at'] as String),
      updatedAt: DateTime.parse(json['updated_at'] as String),
      closedAt: json['closed_at'] != null ? DateTime.parse(json['closed_at'] as String) : null,
      metadata: json['metadata'] as Map<String, dynamic>?,
    );
  }
}
